app/Http/Controllers/Webapp/PostController.php

```diff
@@ -0,0 +1,1890 @@
+                      ->setTagName('button')
+                      ->setRenderSection(RenderableRegistry::SECTION_END)
+                      ->setAttributes([
+                        'class' => 'btn btn-success btn-sm filter-post'
+                      ])
+                  ])
+              ])
+              ,
+            (new TFoot)
+              ->setComponents([
+                (new OneCellRow)
+                  ->setComponents([
+                    (new Pager),
+                    (new HtmlTag)
+                      ->setAttributes(['class' => 'pull-right'])
+                      ->addComponent((new ShowingRecords)->setTemplate('webapp/components/nayjest_showing_records_template')),
+                  ])
+              ])
+              ,
+          ]);
+        $grid = new Grid($cfg);
+        $grid = $grid->render();
+        return view('webapp/posts/index', [
+          'posts_grid' => $grid,
+          'action_delete' => $this->currentUser->is(User::ADMIN) ? 'delete' : 'writerDelete',
+          'is_filter' => count(array_column($request->all(), 'filters')) ? true : false
+        ]);
+    }
+
+    public function getCreate(Request $request) {
+        $categoriesSet = [];
+        $sites = Site::where('status', 1)->get();
+
+        if(count($sites) == 0) {
+            $request->session()->flash('flashMessageSet',
+              (new MessageSet())->setType('warning')->add(new Message('ターゲットサイトがありません。記事を新規作成する前にターゲットサイトを追加してください', 'warning'))
+            );
+
+            return redirect()->to(action('Webapp\\PostController@getIndex'))->send();
+        }
+
+        foreach(Category::all() as $category) {
+            if(!isset($categoriesSet[$category->site_id])) {
+                $categoriesSet[$category->site_id] = [];
+            }
+            $categoriesSet[$category->site_id][] = $category;
+        }
+
+        return view('webapp/posts/create', [
+          'sites' => $sites,
+          'categoriesSet' => $categoriesSet
+        ]);
+    }
+
+    public function postCreate(Request $request) {
+        $postTitle = $request->get('post_title');
+        $postContent = $request->get('post_content');
+        $postDescription = trim($request->get('post_description'));
+        if($request->get('site_id') == ''){
+            $this->_flashMessage($request, 'ターゲットサイトを選んでください', 'danger');
+            return redirect()->to(action('Webapp\\PostController@getIndex'))->send();
+        }
+
+        $post = new Post();
+
+        if(($postId = $request->get('post_id', 0)) != 0) {
+            $post = (!empty($postTmp = Post::find($postId)) && $postTmp->user_id == $this->currentUser->id) ? $postTmp : $post;
+        }
+        $post->title = !empty($postTitle) ? $postTitle : 'タイトルなし';
+        $post->content = !empty($postContent) ? $postContent : '&nbsp;';
+        $post->user_id = $this->currentUser->id;
+        $post->site_id = $request->get('site_id');
+        $post->category = $request->get('category');
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "category" 5 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.relipa.blog/coding_rules#rule_key=php%3AS1192)